controller.py

```
@@ -0,0 +1,201 @@
+        """Extract audio for the selected files."""
+
+        total_files = len(self.selected_files)
+        success_count = 0
+        last_output_path: Optional[str] = None
+        errors: List[str] = []
+
+        if total_files == 0:
+            return ExtractionResult(0, 0, None, [])
+
+        if not self.ffmpeg_path:
+            raise RuntimeError("FFmpeg is not configured.")
```

Ruff (0.13.1)

Avoid specifying long messages outside the exception class

(TRY003)